src/snowcli/cli/nativeapp/manager.py

```diff
@@ -35,3 +66,86 @@ def deploy_root(self) -> Path:
 
     def build_bundle(self) -> None:
         build_bundle(self.project_root, self.deploy_root, self.artifacts)
+
+    def sync_deploy_root_with_stage(self, role: str, app_pkg: str):
+        schema_and_stage: str = self.definition["source_stage"]
+
+        # Does a stage already exist within the app pkg, or we need to create one?
+        # Using "if not exists" should take care of either case.
+        log.info("Checking if stage exists, or creating a new one if none exists.")
+        with self.use_role(role):
+            self._execute_query(
+                f"create schema if not exists {app_pkg}.{schema_and_stage.split('.')[0]}"
+            )
+            self._execute_query(
+                f"""
+                    create stage if not exists {app_pkg}.{schema_and_stage}
+                    encryption = (TYPE = 'SNOWFLAKE_SSE')"""
+            )
+
+        # Perform a diff operation and display results to the user for informational purposes
+        log.info(
+            f"Performing a diff between the Snowflake stage and your local deploy_root {self.deploy_root} directory."
+        )
+        diff: DiffResult = stage_diff(self.deploy_root, f"{app_pkg}.{schema_and_stage}")
+        log.info("Listing results of diff:")
+        print_result(
+            ObjectResult(
+                {
+                    "only on local": ", ".join(diff.only_local),
+                    "only on stage": ", ".join(diff.only_on_stage),
+                    "modified/unknown": ", ".join(diff.different),
+                    "identical": ", ".join(diff.identical),
+                }
+            )
+        )
```

Comment:
Please avoid printing outside the command. Returning an output should be limited to the commands so we have proper control about formatting and behaviour.